data/mesongenbuildreq.py

```diff
@@ -0,0 +1,16 @@
+import subprocess
+import json
+import sys
+deps_json = json.loads(subprocess.run([sys.argv[0], "introspect", "--dependencies", "meson.build"], capture_output=True).stdout)
```

Comment:
`sys.argv[0]` is the name of the script - so you rerun the script in itself, resulting in an endless loop'

Matching `%{shrink: mesongenbuildreq %{__meson} \` you want `sys.argv[1]`